src/parse_type.rs

```diff
@@ -109,6 +110,115 @@ pub(crate) fn consume_generic_params(tokens: &mut TokenIter) -> Option<GenericPa
     })
 }
 
+fn consume_generic_arg(tokens: Vec<TokenTree>) -> GenericArg {
+    // Note: method not called if tokens is empty
+    let mut tokens = tokens.into_iter().peekable();
+
+    // Try parsing 'lifetime
+    if let TokenTree::Punct(punct) = tokens.peek().unwrap() {
+        if punct.as_char() == '\'' {
+            let tk_lifetime = punct.clone();
+            tokens.next(); // consume '
+
+            // after the ', there must be a single identifier
+            match tokens.next() {
+                Some(TokenTree::Ident(ident)) => {
+                    assert!(
+                        tokens.next().is_none(),
+                        "cannot parse lifetime generic argument"
+                    );
+
+                    return GenericArg::Lifetime { tk_lifetime, ident };
+                }
+                Some(other) => {
+                    panic!(
+                        "expected identifier after ' lifetime symbol, got {:?}",
+                        other
+                    );
+                }
+                None => {
+                    panic!("expected identifier after ' lifetime symbol, but ran out of tokens")
+                }
+            }
+        }
+    }
+
+    // Then, try parsing Item = ...
+    // (there is at least 1 token, so unwrap is safe)
+    let before_ident = tokens.clone();
+    if let TokenTree::Ident(ident) = tokens.next().unwrap() {
+        if let Some(TokenTree::Punct(punct)) = tokens.next() {
+            if punct.as_char() == '=' {
+                let remaining: Vec<TokenTree> = tokens.collect();
+
+                return GenericArg::Binding {
+                    ident,
+                    tk_equals: punct,
+                    ty: TyExpr { tokens: remaining },
+                };
+            }
+        }
+    }
```

Comment:
This doesn't cover generic bindings (eg `LendingIterator<Item<'_> = XXX>`), though it doesn't really matter for now.

Reply:
Added `TODO`